pkg/globalnet/controllers/ipam/ippool_test.go

```diff
@@ -0,0 +1,254 @@
+	Context("IP Release", testIpRelease)
+})
+
+const (
+	requestIp1 = "169.254.1.1"
+	service1   = "default/service1"
+	pod1       = "default/pod1"
+	pod2       = "default/pod2"
+	testCidr   = "169.254.1.1/30"
+)
+
+func testPoolCreation() {
+	When("CIDR is missing the prefix", func() {
+		It("should fail to create the IP pool with an error", func() {
+			pool, err := NewIpPool("169.254.0.0")
+			Expect(err).To(HaveOccurred())
+			Expect(pool).To(BeNil())
+		})
+	})
+	When("CIDR prefix is /33", func() {
+		It("should fail to create IP pool with an error", func() {
+			pool, err := NewIpPool("169.254.0.0/33")
+			Expect(err).To(HaveOccurred())
+			Expect(pool).To(BeNil())
+		})
+	})
+	When("CIDR Prefix is /32", func() {
+		It("should fail to create IP pool with an error", func() {
+			pool, err := NewIpPool("169.254.1.0/32")
+			Expect(err).To(HaveOccurred())
+			Expect(pool).To(BeNil())
+		})
+	})
+	When("CIDR Prefix is /31", func() {
+		It("should fail to create IP pool with an error", func() {
+			pool, err := NewIpPool("169.254.1.0/31")
+			Expect(err).To(HaveOccurred())
+			Expect(pool).To(BeNil())
+		})
+	})
+	When("CIDR Prefix is /30", func() {
+		It("should create IP pool of size 2", func() {
+			pool, err := NewIpPool("169.254.1.0/30")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(pool.available)).Should(Equal(2))
+		})
+	})
+	When("CIDR Prefix is /24", func() {
+		It("should create IP pool of size 254", func() {
+			pool, err := NewIpPool("169.254.1.0/24")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(pool.available)).Should(Equal(254))
+		})
+	})
+	When("CIDR Prefix is /16", func() {
+		It("should create an IP pool of size 65534", func() {
+			pool, err := NewIpPool("169.254.1.0/16")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(pool.available)).Should(Equal(65534))
+		})
+	})
+	When("Any pool is created", func() {
+		It("should have allocated pool size of 0", func() {
+			pool, err := NewIpPool("169.254.1.0/16")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(pool.allocated)).Should(Equal(0))
+		})
+	})
+
+}
+
+func testIpAllocation() {
+	When("Any IP is requested", func() {
+		pool, _ := NewIpPool(testCidr)
+		available := len(pool.available)
+		allocated := len(pool.allocated)
+		service1Ip, err := pool.Allocate(service1)
+
+		It("should not return an error", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should return a valid IP for CIDR range", func() {
+			Expect(service1Ip).NotTo(BeNil())
+			_, ipnet, _ := net.ParseCIDR(testCidr)
+			Expect(ipnet.Contains(net.ParseIP(service1Ip))).Should(BeTrue())
+		})
+
+		It("should decrement available IPs pool size by 1", func() {
+			Expect(available - len(pool.available)).Should(Equal(1))
+		})
+
+		It("should increment allocated IPs pool size by 1", func() {
+			Expect(len(pool.allocated) - allocated).Should(Equal(1))
+		})
+
+		It("should not decrement available IPs pool size if same key is reused", func() {
```

Comment:
Nit: Inconsistent naming, this is "IP pool" above and "IPs pool" here (I think the singular makes more sense)